.github/workflows/test.yml

```
@@ -0,0 +1,76 @@
+name: test
+
+on:
+  workflow_dispatch:
+  pull_request:
+    branches:
+      - feature/statistics/impl
+
+jobs:
+  ubuntu-build-test:
+    runs-on: ubuntu-latest
+    environment:
+      name: codecov
+
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          path: src/Fast-DDS
+
+      - uses: ./src/Fast-DDS/.github/actions/install-apt-packages
+      - uses: ./src/Fast-DDS/.github/actions/install-gtest
+      - uses: ./src/Fast-DDS/.github/actions/install-python-packages
+      - uses: ./src/Fast-DDS/.github/actions/fetch-fastdds-repos
+
+      - name: Update colcon mixin
+        run: |
+          colcon mixin add default \
+            https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
+          colcon mixin update default
+        continue-on-error: true
+
+      - name: Build workspace
+        run: |
+          cat src/Fast-DDS/.github/workflows/colcon.meta
+          colcon build \
+            --event-handlers=console_direct+ \
+            --metas src/Fast-DDS/.github/workflows/colcon.meta \
+            --mixin coverage-gcc
+
+      - name: Run tests
+        run: |
+          source install/setup.bash && \
+          colcon test \
```

I think that `colcon test` already takes care of sourcing dependencies...
```suggestion
          colcon test \
```